packages/components/src/components/ConfirmationModal/ConfirmationModal.tsx

```diff
@@ -0,0 +1,62 @@
+import type { DisclosureState } from "ariakit";
+import React from "react";
+import { Button } from "../Button";
+import { Modal, ModalBody, ModalFooter } from "../Modal/index";
+
+export interface ConfirmationModalProps {
+  children: NonNullable<React.ReactNode>;
+  state: DisclosureState;
+  buttonLabel?: string;
+  buttonVariant?: "destructive" | "primary";
+  onConfirm: () => void;
+}
+
+/**
+ * A modal dialog component that forces the user to acknowledge and make a choice
+ * @param props Object with props
+ * @param props.state state to open and close the modal
+ * @param props.buttonVariant variant of the confirm button
+ * @param props.buttonLabel label for the confirm button
+ * @param props.onConfirm function to call when clicking the confirm button
+ * @param props.children React children for the modal body
+ * @returns React.JSX.Element
+ */
+
+const ConfirmationModal = React.forwardRef<
+  HTMLDivElement,
+  ConfirmationModalProps
+>(
+  (
+    {
+      state,
+      buttonVariant = "primary",
+      buttonLabel = "Confirm",
```

Comment:
For the future: Do we want to keep default text on Pluto components? Or, how should we manage them?
We are talking about supporting multiple languages since it is a requisite for blue-collar companies, so it can become a topic soon.

Reply:
Based on my previous comment https://github.com/Localitos/pluto/pull/1107/files#r1392326834 I would also suggest renaming it to `confirmationLabel` or `confirmationText`.

Reply:
> For the future: Do we want to keep default text on Pluto components? Or, how should we manage them?

I don't see a problem with that. If we have a default state of a component it should also have a default label.
When we introduce I18n then we need to adjust for that when we need this feature..